packages/block-library/src/navigation/index.php

```diff
@@ -556,7 +556,7 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 			$is_list_open       = false;
 			$inner_blocks_html .= '</ul>';
 		}
-		if ( 'core/site-title' === $inner_block->name || 'core/site-logo' === $inner_block->name ) {
+		if ( 'core/site-title' === $inner_block->name || ( 'core/site-logo' === $inner_block->name && $inner_block->render() ) ) {
 			$inner_blocks_html .= '<li class="wp-block-navigation-item">' . $inner_block->render() . '</li>';
 		} else {
 			$inner_blocks_html .= $inner_block->render();
```

Comment:
One thing we could do is introduce a temporary var before this conditional to hold the result of `$inner_block->render()` so we don't call it twice (although the PHP interpreter will likely take care of that optimization for us, so it's probably not crucial):

```suggestion
		$inner_block_content = $inner_block->render();
		if ( 'core/site-title' === $inner_block->name || ( 'core/site-logo' === $inner_block->name && $inner_block_content ) ) {
			$inner_blocks_html .= '<li class="wp-block-navigation-item">' . $inner_block_content . '</li>';
		} else {
			$inner_blocks_html .= $inner_block_content;
```